expression/builtin_time_vec.go

```diff
@@ -1203,11 +1203,53 @@ func (b *builtinToSecondsSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 }
 
 func (b *builtinSubDurationAndStringSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSubDurationAndStringSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	if err := b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
+		return err
+	}
+	buf0 := result
+	buf1, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.MergeNulls(buf1)
+	arg0s := buf0.GoDurations()
+	resultSlice := result.GoDurations()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		// calculate
```

Comment:
These comments are meaningless, we can remove it.

Reply:
OK，I have  changed it .